From 79c2d1f73f821fa274a55f683b480e2d77634cf6 Mon Sep 17 00:00:00 2001
From: Yurii Rashkovskii <me@yrashk.com>
Date: Wed, 1 Aug 2018 23:06:23 +0700
Subject: [PATCH] Problem: padding in base32 encoding

I've noticed that currently used Base32 encoding has padding enabled.
While this is not harmful per se, the non-alphanumeric character of "="
might be problematic on some file systems. Besides, there's no
particular need to pad that I can think of as the size of transported
data is typically known in SIT's context (RFC 4648 3.2)

Solution: switch to non-padded version of Base32

Explicitly stating this is valid under RFC 4648 3.2:

   Implementations MUST include appropriate pad characters at the end of
   encoded data *unless* the specification referring to this document
   explicitly states otherwise.

Luckily for us, we've only used 20-byte long Blake2 and SHA-1 hashing
algorithms so far so SIT has never produced padded record names (that
is, unless somebody used SIT with a manually modified config to state
a different length of the Blake2 algorithm output -- which is highly
unlikely at this stage of SIT's adoption).
---
 sit-core/src/encoding.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sit-core/src/encoding.rs b/sit-core/src/encoding.rs
index 165e761..73a7e09 100644
--- a/sit-core/src/encoding.rs
+++ b/sit-core/src/encoding.rs
@@ -26,7 +26,7 @@ impl Deref for Encoding {
 
     fn deref(&self) -> &Self::Target {
         match self {
-            &Encoding::Base32 => &data_encoding::BASE32,
+            &Encoding::Base32 => &data_encoding::BASE32_NOPAD,
         }
     }
 }
-- 
2.16.4

